python_modules/dagster/dagster/core/definitions/sensor_definition.py

```diff
@@ -173,11 +173,11 @@ class SensorDefinition:
 
     def __init__(
         self,
-        name: str,
         evaluation_fn: Callable[
             ["SensorEvaluationContext"],
             Union[Generator[Union[RunRequest, SkipReason], None, None], RunRequest, SkipReason],
         ],
+        name: Optional[str] = None,
```

Comment:
Making this parameter optional would cause a breaking change if someone is directly instantiating a SensorDefinition and passing `name` as a positional arg. One option would be to keep a positional arg for name, like `name_deprecated`, and print a warning if the user is passing name as a positional arg. then deprecate this positional arg later

Reply:
Another option is to make both name and evaluation_fn default to None, but then still hard error if evaluation_fn is None at construction time.

Reply:
yeah i like that option. prevents us from having to do a follow up task